lib/spack/docs/configuring_compilers.rst

```diff
@@ -281,3 +281,30 @@ Mixing Compilers
 ----------------
 
 For more options on configuring Spack to mix different compilers for different languages, see :ref:`the toolchains configuration docs <toolchains>`.
+
+To disable mixing (e.g. if you have multiple compilers defined, but want each concretized DAG to use one of them consistently), you can set:
+
+.. code-block:: yaml
+
+   concretizer:
+     compiler_mixing: false
+
+This affects root specs and any (transitive) link or run dependencies.
+Build-only dependencies are allowed to use different compilers (even when this is set).
+
+.. note::
+
+   Mixing detection is based on identifying compilers by name and version.
+   If you have two compilers that differ only in terms of variants, ``compiler_mixing:false`` would allow using both in a DAG.
```

Comment:
If we go with the implementation based on nodes, we can:

```suggestion
```

Reply:
I tested node-based identity and it ran into https://github.com/spack/spack/pull/51135#discussion_r2437745875, so I think name/version-based identity will be needed.

EDIT: perhaps in https://github.com/spack/spack/pull/51135#discussion_r2445563338 you are suggestion a way to do this with node-identity - I'll check that.

Reply:
To confirm: this passes if I apply this and update the `count`. I'm set up a side meeting to resolve some questions I have about the logic, but something like this will land tomorrow in this PR and it should no longer rely on compiler name/version for identity.